lib/explorer/polars_backend/series.ex

```diff
@@ -193,29 +193,41 @@ defmodule Explorer.PolarsBackend.Series do
   def add(%Series{} = left, %Series{} = right),
     do: Shared.apply_series(left, :s_add, [right.data])
 
-  def add(left, right) when is_number(right), do: apply_scalar_on_rhs(:add, left, right)
-  def add(left, right) when is_number(left), do: apply_scalar_on_lhs(:add, left, right)
+  def add(left, right) when is_number(right) or right in [:nan, :infinity, :neg_infinity],
```

Comment:
What do you think about using `defguardp` to define this? I just can't think of a good name :D

    defguardp is_numerical(n) when is_number(n) or n in [:nan, :infinity, :neg_infinity]

Reply:
I agree. I thought of the name is_number?/1, but it might be confused with Elixir's is_number/1. So I think the name is_numerical/1 is probably better.

WDYT?

Reply:
perfect!

Reply:
Should I create this guard in explorer/series.ex or polars_backend/series.ex? (It can be used in [basic_numeric_operation/3](https://github.com/elixir-nx/explorer/pull/512/commits/de1144a7c35b3a6d78f05f78aeb09203c271a53a#diff-2449d4b4a415673b77c0ab0ba4add0310f4a28d33c3e9a9b06f05cd529166436R2214) and [valid_for_bool_mask_operation?/2](https://github.com/elixir-nx/explorer/pull/512/commits/50fa195fe9ee88cbfe76f519c898258b62e7d2cd#diff-2449d4b4a415673b77c0ab0ba4add0310f4a28d33c3e9a9b06f05cd529166436R2558) functions)

Reply:
Explorer.Series. If you want to share, you can create it Utils, but it is small enough that should be fine to duplicate. We should also rename `numeric_dtype?` to `is_numeric_dtype` since guards have the `is_` prefix instead of question marks.

Reply:
So should I rename the [io_dtype?/1](https://github.com/elixir-nx/explorer/blob/main/lib/explorer/series.ex#L81) and [numeric_or_bool_dtype?/1](https://github.com/elixir-nx/explorer/blob/main/lib/explorer/series.ex#L83) guards as well? (is_io_dtype/1 and is_numeric_or_bool_dtype/1)

Reply:
ideally yes, but since there are several, let's do that in another commit to keep this one focused. :)